### DISCUSSION TITLE
How to set a parameter within RunConfig to required

### DISCUSSION THREAD

**Discussion by eliu-fracta** [dagster-io/dagster#21580] [Q&A] [answered]
for the code below, if I set CleanedDataConfig's file_path to ..., it will error out. How can I properly configure it so that the job will require file_path as an input?

```

class CleanedDataConfig(Config):
    file_path: str = Field(default=..., description="Path to the CSV file")


@asset
def cleaned_data(context, config: CleanedDataConfig):
    file_path = config.file_path
    context.log.info(f"Cleaning data from {file_path}")

    data = pd.read_csv(file_path)

    data_cleaned = data.dropna()

    context.log.info("Data cleaning completed.")

    return data_cleaned

etl_job = define_asset_job(name="etl_job", selection=["cleaned_data"], config=RunConfig(
    ops={
        "cleaned_data": CleanedDataConfig()
    }
).to_config_dict())
```

**Comment by garethbrickman** [comment]
The value for file_path would have to be typed as a string, so `default=...` would need to be `default="..."` if you literally wanted the ellipsis to be the value. Otherwise couldn't you just omit the default parameter entirely? 

For making it required, the [Field class has an is_required parameter](https://docs.dagster.io/_apidocs/config#dagster.Field)

**Reply by eliu-fracta** [reply]
In the documentation, that seems a legacy approach? The new way to do it will be using `pydantic import Field` [doc](https://docs.dagster.io/_apidocs/config#dagster.Config). Addiotionally, I'm using ... instead of "..." to make it a required field based on the [documentation](https://docs.dagster.io/concepts/configuration/advanced-config-types#required-config-fields).

**Reply by garethbrickman** [reply]
Ah you're absolutely right, my apologies for getting mixed up with the legacy approach! The way you've set up `Field()` in your code is valid for the new Pythonic config system. 

I think the issue is inside `etl_job` with how `CleanedDataConfig` gets instantiated within the `RunConfig`. Try this format: 

```python
etl_job = define_asset_job(name="etl_job", selection=["cleaned_data"], config=RunConfig(
    ops={
        "cleaned_data": {
            "config": {
                "file_path": "path/to/your/csvfile.csv"
            }
        }
    }
).to_config_dict())
```

**Reply by eliu-fracta** [reply]
Thanks for answering my question! Is it possible to make the parameter inside RunConfig required as well? 

```
etl_job = define_asset_job(name="etl_job", selection=["cleaned_data"], config=RunConfig(
    ops={
        "cleaned_data": {
            "config": {
                "file_path": "path/to/your/csvfile.csv"  <--- make this required
            }
        }
    }
).to_config_dict())
```

**Reply by garethbrickman** [reply, marked as answer]
It will automatically be required in order for the config to be valid for execution otherwise a hard error will occur. Here's how it looks when attempting to materialize within the Dagster UI: 

![Screenshot 2024-05-06 at 10 26 48](https://github.com/dagster-io/dagster/assets/50636644/487c823e-d414-475e-841c-925f17beebe0)

![Screenshot 2024-05-06 at 10 26 58](https://github.com/dagster-io/dagster/assets/50636644/7dc73052-f7ed-4113-b53f-77a595baa3cc)

**Reply by eliu-fracta** [reply]
thank you!
